rs/bitcoin/kyt/src/fetch.rs

```diff
@@ -74,13 +77,24 @@ pub trait FetchEnv {
         &self,
         txid: Txid,
     ) -> TryFetchResult<impl futures::Future<Output = Result<FetchResult, Infallible>>> {
-        let max_response_bytes = match state::get_fetch_status(txid) {
-            None => INITIAL_MAX_RESPONSE_BYTES,
+        let default_provider = Provider::new_with_default(self.btc_network());
+        let (provider, max_response_bytes) = match state::get_fetch_status(txid) {
+            None => (default_provider, INITIAL_MAX_RESPONSE_BYTES),
             Some(FetchTxStatus::PendingRetry {
                 max_response_bytes, ..
-            }) => max_response_bytes,
+            }) => (default_provider, max_response_bytes),
             Some(FetchTxStatus::PendingOutcall { .. }) => return TryFetchResult::Pending,
-            Some(FetchTxStatus::Error(msg)) => return TryFetchResult::Error(msg),
+            Some(FetchTxStatus::Error(err)) => {
+                if err.is_retriable() {
+                    (
+                        err.get_provider()
+                            .map_or(default_provider, |provider| provider.next()),
+                        INITIAL_MAX_RESPONSE_BYTES,
```

Comment:
Since we know how many bytes was used in the last request, shouldn't we re-use the same value from the beginning? This could avoid an unnecessary retry to `INITIAL_MAX_RESPONSE_BYTES`, when `RETRY_MAX_RESPONSE_BYTES` was actually needed.

Reply:
That's true - unless we don't trust the provider.
If we feel like we generally trust them, which we probably do, we can implement the optimization that Greg suggested.